src/ore/src/task.rs

```diff
@@ -95,15 +95,14 @@ async fn unpack_join_result<T>(res: Result<T, JoinError>) -> T {
         Ok(val) => val,
         Err(err) => match err.try_into_panic() {
             Ok(panic) => std::panic::resume_unwind(panic),
-            Err(e) => {
+            Err(_e) => {
                 // Yield explicitly to the tokio runtime. The runtime
                 // can only shutdown each core thread when the running
                 // task yields. This means we could panic here if the `JoinHandle`
                 // is cancelled because it is owned by a separate core thread,
                 // which is already shutdown. We don't want to panic
                 // spuriously when we are already shutting down.
```

Comment:
I think this comment needs to be updated for the new implementation. There's no panicking here now.